packages/datagrid/src/components/datagrid.component.js

```diff
@@ -0,0 +1,250 @@
+	const Component = Inject.get(getComponent, cellRenderer, DefaultCellRenderer);
+
+	return props => <Component {...props} avroRenderer={avroRenderer} getComponent={getComponent} />;
+}
+
+function getAvroRenderer(avroRenderer) {
+	return {
+		booleanCellRenderer: 'DefaultBooleanCellRenderer',
+		dateCellRenderer: 'DefaultDateCellRenderer',
+		intCellRenderer: 'DefaultIntCellRenderer',
+		stringCellRenderer: 'DefaultStringCellRenderer',
+		...avroRenderer,
+	};
+}
+
+export default class DataGrid extends React.Component {
+	static defaultProps = {
+		cellRenderer: 'DefaultCellRenderer',
+		getPinnedColumnDefsFn: serializer.getPinnedColumnDefs,
+		getColumnDefsFn: serializer.getColumnDefs,
+		getRowDataFn: serializer.getRowData,
+		getValueByCellFn: serializer.getValueByCell,
+		headerHeight: HEADER_HEIGHT,
+		headerRenderer: 'DefaultHeaderRenderer',
+		pinHeaderRenderer: 'DefaultPinHeaderRenderer',
+		rowHeight: ROW_HEIGHT,
+		rowSelection: AG_GRID_DEFAULT_ROW_SELECTION,
+	};
+
+	static propTypes = DATAGRID_PROPTYPES;
+
+	constructor(props) {
+		super(props);
+
+		this.handleKeyboard = this.handleKeyboard.bind(this);
+		this.onFocusedColumn = this.onFocusedColumn.bind(this);
+		this.onFocusedCell = this.onFocusedCell.bind(this);
+		this.onGridReady = this.onGridReady.bind(this);
+		this.setGridInstance = this.setGridInstance.bind(this);
+		this.setCurrentFocusedColumn = this.setCurrentFocusedColumn.bind(this);
+		this.updateStyleFocusColumn = this.updateStyleFocusColumn.bind(this);
+	}
+
+	onGridReady({ api }) {
+		this.gridAPI = api;
+	}
+
+	onFocusedCell({ column, ...rest }) {
+		if (!column) {
+			return;
+		}
+
+		if (column.colId !== this.currentColId || column.pinned) {
+			this.removeFocusColumn();
+		}
+
+		this.setCurrentFocusedColumn(column.colId);
+
+		if (column.pinned) {
+			return;
+		}
+
+		this.updateStyleFocusColumn();
+
+		if (this.props.onFocusedCell) {
+			this.props.onFocusedCell({
+				column,
+				...rest,
+			});
+		}
+	}
+
+	onFocusedColumn(colId) {
+		let selectedRowIndex = 0;
+		if (this.gridAPI.getFocusedCell()) {
+			selectedRowIndex = this.gridAPI.getFocusedCell().rowIndex;
+		}
+
+		if (colId !== this.currentColId) {
+			this.removeFocusColumn();
+		}
+
+		this.setCurrentFocusedColumn(colId);
+		this.gridAPI.setFocusedCell(selectedRowIndex, colId);
+		this.updateStyleFocusColumn();
+		this.props.onFocusedColumn(colId);
+	}
+
+	setCurrentFocusedColumn(colId) {
+		this.currentColId = colId;
+	}
+
+	setGridInstance(gridInstance) {
+		this.gridInstance = gridInstance;
+	}
+
+	removeFocusColumn() {
+		/*
+			This is a bad pratice to manipulate straight the DOM.
+			But we don't have the choice if we want to highlight the column.
+			We have to access to the wrapper element of the cell and header.
+			There is a issue on ag-grid to request a feature like this.
+			https://github.com/ag-grid/ag-grid/issues/2216
+			When Ag-grid implements this feature, we can remove the below code
+		*/
```

Comment:
Please extract that big chuck of comment into the README with a pointer to the code. And add a smaller comment pointing to the longer explanation on the README.